app/Http/Controllers/Api/ExperienceSkillsController.php

```diff
@@ -48,4 +49,70 @@ public function destroy(ExperienceSkill $experienceSkill)
     {
         $experienceSkill->delete();
     }
+
+    public function batchStore(BatchStoreExperienceSkill $request)
+    {
+        $newExperienceSkills = $request->validated();
+        $response = [];
+
+        foreach ($newExperienceSkills as $newExperienceSkill) {
+            // Restore soft deleted experienceSkill if it exists, otherwise create a new one.
+            $softDeletedExperienceSkill = ExperienceSkill::withTrashed()
+                ->where([
+                ['skill_id', $newExperienceSkill['skill_id']],
+                ['experience_id', $newExperienceSkill['experience_id']],
+                ['experience_type', $newExperienceSkill['experience_type']]
+            ])->first();
+            if ($softDeletedExperienceSkill) {
+                $softDeletedExperienceSkill->restore();
+                if ($newExperienceSkill['justification'] !== null && $newExperienceSkill['justification'] !== '') {
+                    $softDeletedExperienceSkill->justification = $newExperienceSkill['justification'];
+                }
+                $softDeletedExperienceSkill->save();
+                array_push($response, $softDeletedExperienceSkill);
+            } else {
+                $experienceSkill = new ExperienceSkill($newExperienceSkill);
+                $experienceSkill->skill_id = $newExperienceSkill['skill_id'];
+                $experienceSkill->experience_id = $newExperienceSkill['experience_id'];
+                $experienceSkill->experience_type = $newExperienceSkill['experience_type'];
+                $experienceSkill->justification = $newExperienceSkill['justification'];
+                $experienceSkill->save();
+                $experienceSkill->fresh();
```

Comment:
May be able to avoid calling ->fresh() inside the loop, and refresh all the objects with a single DB query by with `return new JsonResource($response->fresh());` at the end.

Reply:
This will only work if it's an Eloquent collection, though. Not sure if it's doable here where we're pushing to an empty array.